zerver/views/__init__.py

```diff
@@ -572,6 +572,42 @@ def dev_direct_login(request, **kwargs):
     return HttpResponseRedirect("%s%s" % (settings.EXTERNAL_URI_SCHEME,
                                           request.get_host()))
 
+@csrf_exempt
+@require_post
+@has_request_variables
+def api_dev_android_direct_login(request, username=REQ):
+    # This function allows logging in without a password on Android app and should only be called in development
+    # environments. It may be called if the DevAuthBackend is included in settings.AUTHENTICATION_BACKENDS
+    if ( not dev_auth_enabled()) or settings.PRODUCTION:
+        # This check is probably not required, since authenticate would fail without an enabled DevAuthBackend.
+        return json_error("Dev enviornment not enabled.", data={"reason": "dev_disabled"}, status=403)
+    user_profile = authenticate(username=username)
+    login(request, user_profile)
+    if user_profile is None:
+        if return_data.get("valid_attestation") == True:
+            # We can leak that the user is unregistered iff they present a valid authentication string for the user.
+            return json_error("This user is not registered; do so from a browser.", data={"reason": "unregistered"}, status=403)
+        return json_error("Your username or password is incorrect.", data={"reason": "incorrect_creds"}, status=403)
```

Comment:
It seems like several of the lines above are unnecessary, since I don't think `valid_attestation` is set by anything other than the Google auth code paths.  Can you clean this up?